src/core/meshcop/tcat_agent.cpp

```diff
@@ -413,6 +413,21 @@ Error TcatAgent::HandleSingleTlv(const Message &aIncommingMessage, Message &aOut
         case kTlvDecommission:
             error = HandleDecomission();
             break;
+        case kTlvPing:
+            if (tlv.IsExtended())
```

Comment:
both base and extended format TLVs are allowed for all commands - see spec 13.9.  A sender should use the extended format only if the payload doesn't fit in the base format.

Reply:
I changed the code to handle both possibilities.